packages/firestore/src/lite/reference.ts

```diff
@@ -49,15 +51,40 @@ export interface DocumentData {
 }
 
 /**
- * Update data (for use with {@link @firebase/firestore/lite#(updateDoc:1)}) consists of field paths (e.g.
- * 'foo' or 'foo.baz') mapped to values. Fields that contain dots reference
- * nested fields within the document.
+ * Similar to Typescript's `Partial<T>`, but allows nested fields to be
+ * omitted and FieldValues to be passed in as property values.
  */
-export interface UpdateData {
-  /** A mapping between a dot-separated field path and its value. */
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  [fieldPath: string]: any;
-}
+export type PartialWithFieldValue<T> = T extends Primitive
+  ? T
+  : T extends Map<infer K, infer V>
+  ? Map<PartialWithFieldValue<K>, PartialWithFieldValue<V>>
```

Comment:
I don't think map keys can have FieldValues. It makes no sense to set `{[FieldValue.delete()]: 1}`

Reply:
good catch, done.